### ISSUE TITLE
Updating extended properties

### ISSUE THREAD

**Issue by SebastianBecker2** [otris/ews-cpp#142]
From what I can see, updating extended properties is not implemented yet.

Since I will need it to set the title (part of the complete name, which is read-only itself), I will try to implement it.
Not 100% on how to do it. But it seems extending the ews::update class to accept ews::extended_property next to the regular ews::property might do it.

I'm open for suggestions.

**Comment by bkircher** [comment]
Not sure I remember correctly but adding a new `ews::property` constructor to the overload set should be enough when you are dealing with "normal" types.

…with extended properties yes, probably adding a new c'tor to `ews::update` and make sure that `ews::extended_property::to_xml` or so generates correct XML would be a starting point.

**Comment by SebastianBecker2** [comment]
This issue isn't solved in the merge regarding the update of the physical address of contacts.

I'm spread a little thin right now and can't take this issue on right now.
So if someone else wants to take a shot.

**Comment by m-Schlitzer** [comment]
In the test code, you try to update the `physical_address::key::business`, with a `physical_address::key::home`. 

Changing it from: 
```
auto address = ews::physical_address(ews::physical_address::key::home, "",  
                                     "Duckburg", "", "", "");               
auto prop = ews::property(                                                  
    ews::contact_property_path::physical_address::business::city, address); 
```

To: 

```
  auto address = ews::physical_address(ews::physical_address::key::home, "", 
                                       "Duckburg", "", "", "");              
  auto prop = ews::property(                                                 
      ews::contact_property_path::physical_address::home::city, address);    
```

or 
```
  auto address = ews::physical_address(ews::physical_address::key::business, "", 
                                       "Duckburg", "", "", "");              
  auto prop = ews::property(                                                 
      ews::contact_property_path::physical_address::business::city, address);    
```

fixes that problem. At first I thought it might be because there is no business physical_address existent yet, but you can update the business physical_address without initializing anything first.

I can fix that real quick, if that was the problem with this, let me know!

**Comment by m-Schlitzer** [comment]
Otherwise, the updating of the pysical_address seems to work fine.

**Comment by SebastianBecker2** [comment]
I will have to look into it tomorrow to check.
I will get back to you asap.

**Comment by SebastianBecker2** [comment]
You are absolutely right about the test case. My mistake.

But I'm worried I might have caused a bit of confusion here.
This issue #142, is explicitly about updating `extended_properties`. For example the title of a contact as part of the `CompleteName` attribute. To update it, I would have to do something like this:
```cpp
ews::extended_field_uri::property_type title_prop_type("String");
ews::extended_field_uri::property_tag title_prop_id("14917");
ews::extended_field_uri title_prop_uri(title_prop_id, title_prop_type);
ews::extended_property title_prop(title_prop_uri, { "el presidente" });
ews::update update(title_prop);
```
Though `ews::update` only takes `ews::property` but doesn't accept `ews::extended_property`.


The issue with updating the `PhysicalAddress` of a contact was resolved in issue #141. Well, except the test case that I messed up.
The issue here was with regards to indexed properties. Though it was just missing some property_paths.

Sorry if I caused some confusion with this. Both issues seems pretty similar and came up at the same time.
